templates/deploy_fortigate_autoscale.hybrid_licensing.json

```
@@ -314,6 +314,22 @@
                 "description": "Name of subnet 4, a subnet that will gain access from the FortiGate. When creating a new VNet that contains the 'Subnet 4', the input value will be used as the Subnet 4 name, or if left empty, a default name will be used. When using an existing Net, a valid non-empty input will assume the association of the target subnet with the Autoscale, and the target subnet will be associated as 'Subnet 4'; empty value will assume the Autoscale will deploy without using the 'Subnet 4'."
             }
         },
+        "TerminateUnhealthyVM": {
+            "defaultValue": "no",
+            "allowedValues": ["no", "yes"],
+            "type": "String",
+            "metadata": {
+                "description": "Default 'no'. When Set to 'no' VM termination and health will be determined by the Azure health probe and AutoHealing. This is the recommended value. When set to true, the Azure Function code will attempt to determine the health of the VM and will delete unhealthy VMs. "
+            }
+        },
+        "SyncRecoveryCount": {
+            "defaultValue": 3,
+            "minValue": 3,
+            "type": "Int",
+            "metadata": {
+                "description": "When 'Terminate Unhealthy VM' is set to false, the  'Sync Recovery Count' determines the amount of in-sync heartbeats needed to return a VM to a healthy state from an out-of-sync state. "
```

Can use exactly the same description as the AWS one because the descriptions there have been reviewed and approved by Kimberly. We can reuse it. See: https://github.com/fortinet/fortigate-autoscale-aws/pull/69/files#diff-f29189982c71c174a00ebb10838d0fcbfa1a80e9826339eeb84176481a33494cR500

```suggestion
                "description": "Number of consecutive on-time heartbeats required for a VM to become healthy again. This parameter is only used when 'Terminate unhealthy VM' is set to ‘no’ and allows for the VM to recover from an unhealthy state."
```

---

This is different from the original one I saw. I think this could still use some clarification since it's sort of unclear what it does, which if I am correct is: it returns the state to healthy, but just in the logs. There is no action, termination etc related to this parameter.

---

The number is a threshold. It is a counter that will count before the Autoscale changes the vm state from unhealthy to healthy.
It's correct that there will be no action but just the change of the health state. The state means its eligibility to become an elected primary.
You are also correct at the point: No termination.

---

```suggestion
                "description": "When 'Terminate unhealthy VM' is set to ‘no’, this parameter specifies the number of consecutive on-time heartbeats required for the VM to recover from an unhealthy state."
```

---

@klc178 Please also revise it against the description here in the latest release (on AWS): https://github.com/fortinet/fortigate-autoscale-aws/blob/3.4.0/templates/autoscale-new-vpc.template.yaml#L509

This feature is an Autoscale feature generally available for all platforms. We should keep the consistent description for this parameter so let me know if the description needs to update on the AWS project too?

same comment as Termination Unhealthy VM but the link is different. :)